core/src/aggregator.rs

```diff
@@ -81,13 +86,100 @@ impl Aggregator {
             .unwrap();
 
         Ok(Aggregator {
+            db,
             config,
             nofn_xonly_pk,
             verifier_clients,
             operator_clients,
         })
     }
 
+    /// Creates a stream of nonces from verifiers.
+    /// This will automatically get's the first response from the verifiers.
+    pub async fn create_nonce_streams(
+        &self,
+        num_nonces: u32,
+    ) -> Result<
+        (
+            Vec<clementine::NonceGenFirstResponse>,
+            Vec<BoxStream<Result<MuSigPubNonce, BridgeError>>>,
+        ),
+        BridgeError,
+    > {
+        // generate nonces from all verifiers
+        let mut nonce_streams = try_join_all(self.verifier_clients.iter().map(|client| {
```

Comment:
I think this message passing steps should be on the RPC crate. But the things like pubnonce extraction can stay here.

Reply:
Although it shouldn't use types like `clementine::nonce_gen_response::Response` because they are native to RPC crate.